### PR TITLE
Fix client failure when passwords with special characters provided

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose
Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/4110
## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/build.gradle [hunk 1]
```diff
@@ -166,7 +166,7 @@ task startBallerinaSTS() {
             if (!stdOut.toString().contains("ballerina-sts")) {
                 println "Starting Ballerina STS."
                 exec {
-                    commandLine 'sh', '-c', "docker run --rm -d -p 9445:9445 -p 9444:9444 --name ballerina-sts ldclakmal/ballerina-sts:latest"
+                    commandLine 'sh', '-c', "docker run --rm -d -p 9445:9445 -p 9444:9444 --name ballerina-sts sabthar/ballerina-sts:latest"
```
Comment (dilanSachi) [inline ballerina/build.gradle:169]: Shall we add the github url of the docker image as well? Here or in the docker hub readme file.

File: ballerina/client_oauth2_provider.bal [hunk 2]
```diff
@@ -381,6 +384,16 @@ isolated function getAccessTokenFromTokenRequestForPasswordGrant(PasswordGrantCo
     return accessToken;
 }
 
+isolated function getEncodedUsernamePassword(string username, string password) returns [string,string]|Error {
+    do {
+        string encodedUserName = check url:encode(username, "UTF-8");
```
Comment (dilanSachi) [inline ballerina/client_oauth2_provider.bal:389]: shall we use a constant for `utf-8`

File: ballerina/build.gradle (hunk 1 above)
Comment (MohamedSabthar) [reply ballerina/build.gradle:169]: added docker hub readme link

File: build.gradle [hunk 3]
```diff
@@ -30,6 +30,7 @@ ext.stdlibCryptoVersion = project.stdlibCryptoVersion
 ext.stdlibLogVersion = project.stdlibLogVersion
 ext.stdlibRegexVersion = project.stdlibRegexVersion
 ext.stdlibTimeVersion = project.stdlibTimeVersion
+ext.stdlibUrlVersion = project.stdlibUrlVersion
 // Transitive dependencies
 ext.stdlibTaskVersion = project.stdlibTaskVersion
 ext.stdlibConstraintVersion = project.stdlibConstraintVersion
```
Comment (ThisaruGuruge) [inline build.gradle:36]: We do not need these `ext` variable definitions in the `build.gradle` file

Comment (MohamedSabthar) [reply build.gradle]: Removed with [59fd3b5](https://github.com/ballerina-platform/module-ballerina-oauth2/pull/915/commits/59fd3b522b618d75e07fe3a8fa95b36c724d9dbe)

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**